### DISCUSSION TITLE
File uploaded using upload method of file upload library always show content-type as 'application/octet-stream'

### DISCUSSION THREAD

**Discussion by vaivk369** [Sunbird-Knowlg/Community#154] [Bugs]
In coKreat, we are using "@project-sunbird/sunbird-file-upload-library": "1.0.2".  The files uploaded using this library are always show content-type as 'application/octet-stream'  in the response headers. 
 
 In  the file, https://github.com/Sunbird-Knowlg/sunbird-file-upload-library/blob/main/lib/azure.js, the following code is hard coded. 
 
```
  "headers": {
                    "x-ms-blob-type": "BlockBlob",
                    'Content-Type': 'application/octet-stream'
                },
```
The content type of the file should be passed instead of hard coded one. 

[@pallakartheekreddy](https://github.com/pallakartheekreddy), [@maheshkumargangula](https://github.com/maheshkumargangula)
CC: @vinukumar-vs

**Comment by HarishGangula** [comment]
@vaivk369  Can you share complete details of it, by default `Content-Type` will be `application/octet-stream` for azure blob uploads but we can enhance to take the file content type in the next version. Meanwhile, you can use chunk upload with options as a workaround.
cc: @pallakartheekreddy

**Comment by vaivk369** [comment]
@HarishGangula the upload function has been used in various places in coKreat and also in workspace of ED portal. If possible can you provide the hotfix for the same?

**Reply by Krishnaj20** [reply]
@pallakartheekreddy - this is blocking coKreat 6.0.0 release. Request you to co-ordinate with @vinukumar-vs and @vaivk369 to close this on priority. Thanks

**Reply by HarishGangula** [reply]
@vaivk369 There is new version("@project-sunbird/sunbird-file-upload-library": "1.0.3") of the package that is updated with required changes 
please use this version.
cc: @pallakartheekreddy

**Reply by vaivk369** [reply]
Thank you @HarishGangula . Will use 1.0.3 and let you know.

**Reply by vaivk369** [reply]
Updating file library to 1.0.3 has fixed the issue on coKreat. Thank you.

@pallakartheekreddy @HarishGangula  we also need to update the     "@project-sunbird/sunbird-file-upload-library": "1.0.3", in collection editor. Can we please give us updated version of collection editor as well?

**Reply by HarishGangula** [reply]
@vaivk369  Collection editor is updated, take the latest version of it.

**Reply by vinukumar-vs** [reply]
This worked..

@pallakartheekreddy Hope the release notes also updated on Knowlg. Please share.

@vaivk369 Update the same on CoKreat release notes as well..

@pallakartheekreddy @HarishGangula Thanks for unblocked the blocker issue of coKreat release-6.0 sign-off.

**Reply by pallakartheekreddy** [reply]
@vinukumar-vs Release notes already updated.

**Reply by vaivk369** [reply]
The release notes for Knowlg https://knowlg.sunbird.org/use/release-notes/release-5.5.0 and same has been updated in the coKreat.

**Comment by Aparna-karri591** [comment]
@HarishGangula, @vaivk369 can you please let us know is there any impacted area which we need to test as part of this fix on coKreat and sunbirdED

cc: @vinukumar-vs @pallakartheekreddy @ashokreddy1208 [@PoojaJManjooran](https://github.com/PoojaJManjooran)

**Comment by vaivk369** [comment]
@Aparna-karri591 file uploads and print preview in coKreat.

**Reply by Aparna-karri591** [reply]
is there any impact On sunbirdED ?

**Comment by ashokreddy1208** [comment]
@HarishGangula @vaivk369 @pallakartheekreddy do we need to deploy this changes on sunbird ED aswell? can you please confirm .

cc @vinukumar-vs @rajeevsathish

**Reply by pallakartheekreddy** [reply]
@vaivk369 @vinukumar-vs Please update on @ashokreddy1208 query.

**Comment by vaivk369** [comment]
We do not have to make changes in ED-portal as no flow is breaking due to this issue.
